cnxarchive/database.py

```diff
@@ -186,9 +196,14 @@ def get_collections(module_ident, plpy):
         FROM trees c JOIN t ON (c.nodeid = t.parent)
         WHERE not c.nodeid = ANY(t.path)
     )
-    SELECT DISTINCT m.module_ident
-    FROM t JOIN latest_modules m ON (t.document = m.module_ident)
-    WHERE t.parent IS NULL ORDER BY m.module_ident
+    SELECT * FROM (
+        SELECT DISTINCT m.module_ident, m.revised,
+            MAX(m.revised) OVER (PARTITION BY m.uuid) as latest
+        FROM t, modules m
+        WHERE t.parent IS NULL and t.document = m.module_ident
+    ) r
+    WHERE r.revised = r.latest
+    ORDER BY r.module_ident
```

Comment:
no, I think what's happening is that the PARTITION OVER doesn't access the whole table, just the subset that matches the WHERE condition - i.e. the collections that this page is in. The newer ones are not present in the subset - PARITION BY doesn't bring in new rows, it only GROUPs what's there (without collapsing the group to  single row, if you get what I mean)